src/panels/lovelace/editor/conditions/types/ha-card-condition-location.ts

```diff
@@ -0,0 +1,106 @@
+import { LitElement, css, html } from "lit";
+import { customElement, property } from "lit/decorators";
+import { array, assert, literal, object, string } from "superstruct";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import "../../../../../components/ha-check-list-item";
+import "../../../../../components/ha-switch";
+import "../../../../../components/ha-list";
+import type { HomeAssistant } from "../../../../../types";
+import type { LocationCondition } from "../../../common/validate-condition";
+import "../../../../../components/ha-form/ha-form";
+
+const locationConditionStruct = object({
+  condition: literal("location"),
+  locations: array(string()),
+});
+
+const SCHEMA = [
+  {
+    name: "locations",
+    selector: {
+      state: {
+        entity_id: "person.whomever",
+        hide_states: ["unavailable", "unknown"],
+      },
+    },
+  },
+];
+
+@customElement("ha-card-condition-location")
+export class HaCardConditionLocation extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public condition!: LocationCondition;
+
+  @property({ type: Boolean }) public disabled = false;
+
+  public static get defaultConfig(): LocationCondition {
+    return { condition: "location", locations: [] };
+  }
+
+  protected static validateUIConfig(condition: LocationCondition) {
+    return assert(condition, locationConditionStruct);
+  }
+
+  protected render() {
+    return html`
+      <ha-form
+        .hass=${this.hass}
+        .data=${{ locations: this.condition?.locations?.[0] }}
```

Comment:
Would have like to use multi-state selector here, but gated by https://github.com/home-assistant/frontend/pull/25716.